Sources/Overture/With.swift

```diff
@@ -43,7 +66,20 @@ public func with<A>(_ a: A, _ f: (inout A) throws -> Void) rethrows -> A {
 /// - Returns: The result of `f` applied to `a`.
 /// - Note: This function is commonly seen in operator form as "pipe-forward", `|>`.
 @discardableResult
-public func with<A: AnyObject>(_ a: A, _ f: (A) throws -> Void) rethrows -> A {
-  try f(a)
+public func update<A: AnyObject>(reference a: A, _ fs: ((A) -> Void)...) -> A {
```

Comment:
This overload is also for fixing ambiguity with the `inout` version when using inline closures...